### DISCUSSION TITLE
Path to Arduino Audio-Tools Libraries

### DISCUSSION THREAD

**Discussion by HobmanX** [pschatzmann/arduino-audio-tools#1507] [Q&A] [answered]
I installed the Arduino Audio-Tools library in my sketchbook libraries folder using Git Clone "URL address" and did updates by doing git pull and it is working but I notice....... the pathname to AudioTools.h is.....
D:\abc\Documents\Arduino\Sketchbook\libraries\arduino-audio-tools\src\AudioTools.h
I was trying to use filters for one of my sketches.... I can get the example for the FIR filter to work but I was trying to get the LowPassFilter to work but I am getting the following error...........
 error: 'audio_tools::LowPassFilter<T>::LowPassFilter(float, float, float) [with T = float]' is private within this context

d:\abc\Documents\Arduino\Sketchbook\libraries\arduino-audio-tools\src/AudioFilter/Filter.h:319:3: note: declared private here
   LowPassFilter(float frequency, float sampleRate, float q = 0.7071f)
   ^~~~~~~~~~~~~

exit status 1

Compilation error: 'audio_tools::LowPassFilter<T>::LowPassFilter(float, float, float) [with T = float]' is private within this context

Why does it appear to be using 2 different folders......???........ or what else could be wrong...??

**Comment by pschatzmann** [comment, marked as answer]
I think something went wrong with your git pull since you still have some old code: do

    cd \abc\Documents\Arduino\Sketchbook\libraries\arduino-audio-tools
    git pull

**Comment by HobmanX** [comment]
Yes, that is it working now....it is finding the LowPassFilter now and no errors.
Your git pull and folder path is same as what I had been using and I did i recently..... but it sis working now.
Referring to my last post...I posted wrongly.... the Audiotools.h files is in the correct folder in sketchbook libraries etc but it is another file.. ........C:\Users\abc\AppData\Local\Temp\arduino-language-server3805895114\build\sketch\Array256X.h  ...... that made me wonder what was happening. The actual file Array256X.h is a text file containing different arrays of sounds and it definately is in my sketch folder in my sketchbook folder.....Yet when I hover over it in the IDE editor it tells me it is using a copy of it in AppData on C drive.....is that normal..??

Another thing regarding the LowPassFilter....it has 3 parameters, frequency....is this the corner frequency.?? ,   ...samplerate..OK, and a 3rd one called "q".... listed and should be 0.707999 when I hover over it, but it is at 0.5f in the sketch....... Is this the gain at the corner freq..?? should I set it at 0.707999 or leave it at 0.5..?
also the FIR filter has 5 coefficients... where do I find the detail about setting them..??
